pkg/test/framework/components/echo/common/call.go

```diff
@@ -182,8 +182,15 @@ func ForwardEcho(srcName string, clientProvider EchoClientProvider, opts *echo.C
 }
 
 func fillInCallOptions(opts *echo.CallOptions) error {
-	if opts.Target != nil {
-		targetPorts := opts.Target.Config().Ports
+	if opts.Target != nil && opts.TargetWorkload != nil {
```

Comment:
There's a lot of branching here. Can we create 2 functions that handle either Target or TargetWorkload?

Reply:
A bit worried about duplicating all the code which his pretty complex